### ISSUE TITLE
Error Starting GRPC.Server.Supervisor with Phoenix 1.7.0-rc-0

### ISSUE THREAD

**Issue by cdesch** [elixir-grpc/grpc#287]
**Describe the bug**
Following the [`helloworld`](https://github.com/elixir-grpc/grpc/tree/master/examples/helloworld) example and adding it to a Phoenix `1.7.0-rc-0` application, I get the following Error:

```
** (Mix) Could not start application phx_one_seven_grpc: PhxOneSevenGrpc.Application.start(:normal, []) returned an error: shutdown: failed to start child: GRPC.Server.Supervisor
    ** (EXIT) an exception was raised:
        ** (FunctionClauseError) no function clause matching in GRPC.Server.Supervisor.init/1
            (grpc 0.5.0) lib/grpc/server/supervisor.ex:41: GRPC.Server.Supervisor.init([endpoint: Helloworld.Endpoint, port: 50051])
            (stdlib 3.17) supervisor.erl:330: :supervisor.init/1
            (stdlib 3.17) gen_server.erl:423: :gen_server.init_it/2
            (stdlib 3.17) gen_server.erl:390: :gen_server.init_it/6
            (stdlib 3.17) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
```

**To Reproduce**

1) Clone this repo [phx_one_seven_grpc](https://github.com/cdesch/phx_one_seven_grpc) which contains the Phoenix `1.7.0-rc-0` project with `elixir-grpc` installed.
2) Run `iex -S mix` to run the Phoenix Server
3) Error will reproduce

**Expected behavior**
~A clear and concise description of what you expected to happen.~

Expected that the `grpc` server will start with Phoenix

**Logs**
~If applicable, add logs to help explain your problem.~
```
cj@cj-Z:~/projects/phx_one_seven_grpc|main ⇒  iex -S mix
Erlang/OTP 24 [erts-12.2.1] [source] [64-bit] [smp:16:16] [ds:16:16:10] [async-threads:1] [jit]

[notice] Application phx_one_seven_grpc exited: PhxOneSevenGrpc.Application.start(:normal, []) returned an error: shutdown: failed to start child: GRPC.Server.Supervisor
    ** (EXIT) an exception was raised:
        ** (FunctionClauseError) no function clause matching in GRPC.Server.Supervisor.init/1
            (grpc 0.5.0) lib/grpc/server/supervisor.ex:41: GRPC.Server.Supervisor.init([endpoint: Helloworld.Endpoint, port: 50051])
            (stdlib 3.17) supervisor.erl:330: :supervisor.init/1
            (stdlib 3.17) gen_server.erl:423: :gen_server.init_it/2
            (stdlib 3.17) gen_server.erl:390: :gen_server.init_it/6
            (stdlib 3.17) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
[notice] Application google_protos exited: :stopped
[notice] Application protobuf exited: :stopped
[notice] Application plug_cowboy exited: :stopped
[notice] Application cowboy_telemetry exited: :stopped
[notice] Application gettext exited: :stopped
[notice] Application telemetry_poller exited: :stopped
[notice] Application finch exited: :stopped
[notice] Application nimble_options exited: :stopped
[notice] Application nimble_pool exited: :stopped
[notice] Application mint exited: :stopped
[notice] Application hpax exited: :stopped
[notice] Application swoosh exited: :stopped
[notice] Application jason exited: :stopped
[notice] Application xmerl exited: :stopped
[notice] Application tailwind exited: :stopped
[notice] Application esbuild exited: :stopped
[notice] Application phoenix_live_dashboard exited: :stopped
[notice] Application telemetry_metrics exited: :stopped
[notice] Application heroicons exited: :stopped
[notice] Application phoenix_live_view exited: :stopped
[notice] Application phoenix_live_reload exited: :stopped
[notice] Application file_system exited: :stopped
[notice] Application phoenix_html exited: :stopped
[notice] Application postgrex exited: :stopped
[notice] Application ecto_sql exited: :stopped
[notice] Application db_connection exited: :stopped
[notice] Application connection exited: :stopped
[notice] Application phoenix_ecto exited: :stopped
[notice] Application ecto exited: :stopped
[notice] Application decimal exited: :stopped
[notice] Application phoenix exited: :stopped
[notice] Application castore exited: :stopped
[notice] Application websock_adapter exited: :stopped
[notice] Application websock exited: :stopped
[notice] Application phoenix_template exited: :stopped
[notice] Application phoenix_pubsub exited: :stopped
[notice] Application plug exited: :stopped
[notice] Application telemetry exited: :stopped
[notice] Application plug_crypto exited: :stopped
[notice] Application mime exited: :stopped
[notice] Application eex exited: :stopped
[notice] Application grpc exited: :stopped
[notice] Application gun exited: :stopped
[notice] Application cowboy exited: :stopped
[notice] Application ranch exited: :stopped
[notice] Application cowlib exited: :stopped
[notice] Application runtime_tools exited: :stopped
** (Mix) Could not start application phx_one_seven_grpc: PhxOneSevenGrpc.Application.start(:normal, []) returned an error: shutdown: failed to start child: GRPC.Server.Supervisor
    ** (EXIT) an exception was raised:
        ** (FunctionClauseError) no function clause matching in GRPC.Server.Supervisor.init/1
            (grpc 0.5.0) lib/grpc/server/supervisor.ex:41: GRPC.Server.Supervisor.init([endpoint: Helloworld.Endpoint, port: 50051])
            (stdlib 3.17) supervisor.erl:330: :supervisor.init/1
            (stdlib 3.17) gen_server.erl:423: :gen_server.init_it/2
            (stdlib 3.17) gen_server.erl:390: :gen_server.init_it/6
            (stdlib 3.17) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
```

~**Protos**~
~If applicable, add protos to help explain your problem.~

**Versions:**
- OS: Ubuntu 22.04 (5.15.0-53-generic #59-Ubuntu SMP Mon Oct 17 18:53:30 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux)
- Erlang: Erlang/OTP 24 [erts-12.2.1] [source] [64-bit] [smp:16:16] [ds:16:16:10] [async-threads:1] [jit]
- Elixir: Elixir 1.14.1 (compiled with Erlang/OTP 24)
- mix.lock(grpc, gun, cowboy, cowlib): 
```
 "cowboy_telemetry": {:hex, :cowboy_telemetry, "0.4.0", "f239f68b588efa7707abce16a84d0d2acf3a0f50571f8bb7f56a15865aae820c", [:rebar3], [{:cowboy, "~> 2.7", [hex: :cowboy, repo: "hexpm", optional: false]}, {:telemetry, "~> 1.0", [hex: :telemetry, repo: "hexpm", optional: false]}], "hexpm", "7d98bac1ee4565d31b62d59f8823dfd8356a169e7fcbb83831b8a5397404c9de"},
  "cowlib": {:hex, :cowlib, "2.11.0", "0b9ff9c346629256c42ebe1eeb769a83c6cb771a6ee5960bd110ab0b9b872063", [:make, :rebar3], [], "hexpm", "2b3e9da0b21c4565751a6d4901c20d1b4cc25cbb7fd50d91d2ab6dd287bc86a9"},
"grpc": {:hex, :grpc, "0.5.0", "a44cb306625a52fa31a2189ce91b40d24e82569568f0cc214c1e1e0faf54f58a", [:mix], [{:cowboy, "~> 2.9", [hex: :cowboy, repo: "hexpm", optional: false]}, {:cowlib, "~> 2.11", [hex: :cowlib, repo: "hexpm", optional: false]}, {:gun, "~> 2.0.1", [hex: :grpc_gun, repo: "hexpm", optional: false]}], "hexpm", "17b98593fdb1a65be7b2722821266627b3f2fba29bbbd7d0945389427c0d0d5f"},
  "gun": {:hex, :grpc_gun, "2.0.1", "221b792df3a93e8fead96f697cbaf920120deacced85c6cd3329d2e67f0871f8", [:rebar3], [{:cowlib, "~> 2.11", [hex: :cowlib, repo: "hexpm", optional: false]}], "hexpm", "795a65eb9d0ba16697e6b0e1886009ce024799e43bb42753f0c59b029f592831"},
```

**Additional context**

**Comment by polvalente** [comment]
Things should work if you use the github master branch instead of the Hex 0.5.0 version.
Basically, we deprecated the global environment config in favor of a options-based config.

The issue is that you are using the new way in you application tree, but the code expects the old way.

Thanks for the report!

**Comment by polvalente** [comment]
You can either reference git or follow the instructions for 0.5 instead and it should be good to go.

Yes, everything after 0.5 on main will be eventually released on 0.6 :)
